### ISSUE TITLE
Debugging R in Visual Studio not yet implemented

### ISSUE THREAD

**Issue by AndyLuckert** [MikhailArkhipov/vscode-r#39]
Dear community,

**in this project, the README.md states that implementing a debugger is on the To-Do-List.
Are there any ongoing attempts? I would really love to be able to debug R in Visual Studio**.

Usually, I program in Python. Now with my R-script, I did the same steps as usual:
Setting a breakpoint at a line of desire, and simply press F5 to start the debugging process.
Unfortunately, the debug-config file "launch.json" doesn't comprise of a R-related dictionary entry such as e.g. for python:

```
        {
            "name": "Python: Current File (Integrated Terminal)",
            "type": "python",
            "request": "launch",
            "program": "${file}",
            "console": "integratedTerminal"
        }
```

If I imitate the above code-lines and exchange "python" with "R", it says there is no debugger installed named like that.
Searching the vscode market place for R-debugging doesn't seem to offer anything, which obviously contains a debugger.

The closest I found was the extension [R LSP Client](https://marketplace.visualstudio.com/items?itemName=REditorSupport.r-lsp).
Nevertheless, after installing it, there was no way for me to add an R-debugger to my launch.json-file,
since there doesn't appear anything related to R in the available options when clicking the Tab "Debug" > "Add configuration ...".

The simple way the debugging process is described [here](https://docs.microsoft.com/en-us/visualstudio/rtvs/debugging-r-in-visual-studio?view=vs-2017) doesn't work for me since when starting a debugging process via F5, vscode selects one of the available debuggers (I have various for python available, such as "Current File", "Remote", "Flask", Django",.. BUT NOTHING FOR R).

I'd be very happy to get some help, since I really don't want to install RStudio on my system just because I can't debug R-codes in Visual Studio.

Thank you very much in advance!
Kind regards,
Andreas

PS:
The R-version I use is the following:
_R version 3.6.2 (2019-12-12) -- "Dark and Stormy Night"
Copyright (C) 2019 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu (64-bit)_

**Comment by MikhailArkhipov** [comment]
Did you mean Visual Studio or VS Code? R can be debugged in VS 2017 in box and in 2019 via R Tools extension (since RTVS is no longer in box with VS).

**Comment by AndyLuckert** [comment]
I'm referring to the application Visual Studio Code from the Ubuntu repository:
http://packages.microsoft.com/repos/vscode stable/main amd64 Packages

When typing "code --version" in my bash-terminal:
_1.41.0
9579eda04fdb3a9bba2750f15193e5fafe16b959
x64_

What do you mean by saying "box", is that an extension?

I've installed the extension [R Tools](https://marketplace.visualstudio.com/items?itemName=Mikhail-Arkhipov.r), but there is no debugging included, as it is stated even the proper README.md - file.

As for the version of the app (2017 and 2019), I've no idea where to exactly determine from which year my visual studio code is, but I installed it this very year 2019, so I suppose it's 2019, the most recent.

**Comment by MikhailArkhipov** [comment]
So you do mean [Visual Studio Code](https://code.visualstudio.com/) and not [Visual Studio](https://visualstudio.microsoft.com/). Asked b/c you said "I would really love to be able to debug R in Visual Studio".

Correct, there is no debugging in VS Code with R. I am the only one occasionally working on this project at my own time.

**Comment by AndyLuckert** [comment]
Ok, I'm sorry for this naming confusion.
I've just learned about this distinction myself to be honest. As I'm a Linux Lubuntu - user, there is not Visual Studio full-fledged IDE for my system, just VS CODE.

See the following link for those, who are interested:
https://stackoverflow.com/questions/30527522/what-are-the-differences-between-visual-studio-code-and-visual-studio

Alright, then I hope you'll keep up your motivation as a lone-wolf on your mission, for the sake of all people like me who'd like to use R to its fullest in VS Code!
What do you estimate about when a first release of an R-debugger might appear in the VS CODE marketplace? ;)

**Comment by MikhailArkhipov** [comment]
Yes, connecting existing RTVS debugger to VS Code is not yet done.

**Comment by AndreasLuckert** [comment]
Alright Mikhail, keep us posted :)
